src/components/Collection/Tags/TagDetailsModal.tsx

```diff
@@ -1,20 +1,81 @@
-import React from 'react';
+import React, { useMemo, useRef } from 'react';
 import { Link } from 'react-router-dom';
-import { mdiOpenInNew } from '@mdi/js';
+import { mdiLoading, mdiOpenInNew } from '@mdi/js';
 import Icon from '@mdi/react';
+import { useVirtualizer } from '@tanstack/react-virtual';
 import cx from 'classnames';
+import { debounce } from 'lodash';
 
 import AnidbDescription from '@/components/Collection/AnidbDescription';
 import ModalPanel from '@/components/Panels/ModalPanel';
 import { useFilteredSeriesInfiniteQuery } from '@/core/react-query/filter/queries';
 import useFlattenListResult from '@/hooks/useFlattenListResult';
 
+import type { SeriesType } from '@/core/types/api/series';
 import type { TagType } from '@/core/types/api/tags';
 
-const TagDetailsModal = React.memo(({ onClose, show, tag }: { show: boolean, tag?: TagType, onClose: () => void }) => {
+const SeriesLink = React.memo(({ extraPadding, series }: { series: SeriesType, extraPadding: boolean }) => (
+  <Link
+    to={`/webui/collection/series/${series.IDs.ID}`}
+    className={cx(
+      'flex justify-between align-middle hover:text-panel-text-primary',
+      extraPadding && ('pr-4'),
+    )}
+  >
+    <span
+      className="line-clamp-1"
+      data-tooltip-id="tooltip"
+      data-tooltip-content={series.Name}
+      data-tooltip-delay-show={500}
+    >
+      {series.Name}
+    </span>
+    <Icon path={mdiOpenInNew} size={1} className="shrink-0" />
+  </Link>
+));
+
+const SeriesVirtualizer = (
+  { data, dataSize, fetchNext }: { data: SeriesType[], dataSize: number, fetchNext: () => void },
+) => {
+  const scrollRef = useRef<HTMLDivElement>(null);
+  const virtualizer = useVirtualizer({
+    count: dataSize,
+    getScrollElement: () => scrollRef.current,
+    getItemKey: (idx) => {
+      const seriesId = data[idx]?.IDs.ID;
+      return seriesId ? `tag-series-id-${seriesId}` : `tag-series-idx-${idx}`;
+    },
+    estimateSize: () => (25.6), // Standard line height
+    gap: 8,
+    overscan: 12,
+  });
+
+  return (
+    <div className="shoko-scrollbar max-h-[12.5rem] overflow-y-auto" ref={scrollRef}>
+      <div className="relative" style={{ height: virtualizer.getTotalSize() }}>
+        {virtualizer.getVirtualItems().map(({ index, key, size, start }) => {
+          const series = data?.[index] ?? fetchNext();
```

Comment:
```suggestion
          const series = data[index];
          if (!series) fetchNext();
```
The IDE was complaining about void function being assigned, also data is never undefined, it'll at least be an empty array